### PR TITLE
modulized movie-view and moved in lib/movie

### PR DESCRIPTION
**Pull request by jenna1k**

### CODE REVIEW

File: apps/catalog-marketplace/catalog-marketplace/src/app/movie/view/view.component.html [hunk 1]
```diff
@@ -20,43 +20,10 @@ <h3 class="mat-body-1">a film by {{ director.firstName }} {{ director.lastName }
...
-              <ul>
-                <li fxLayout="row"><b>Beginning of
-                    rights</b><span>{{ movie.salesAgentDeal.rights.from | date:'LLL-dd, yyyy' }}</span></li>
-                <li fxLayout="row"><b>End of
-                    rights</b><span>{{ movie.salesAgentDeal.rights.to | date:'LLL-dd, yyyy' }}</span></li>
-              </ul>
-            </article>
-            <mat-divider [inset]="true"></mat-divider>
-          </ng-container>
+          <movie-display-availabilities [movie]="movie"></movie-display-availabilities>
```
Comment (Maxstgt) [inline apps/catalog-marketplace/catalog-marketplace/src/app/movie/view/view.component.html]: if it is possible, try to pass down the value only that you need. `<movie-display-availabilities [movie]="movie.sales.AgentDeal.rights.from"></movie-display-availabilities>`

File: apps/catalog-marketplace/catalog-marketplace/src/app/movie/view/view.module.ts [hunk 2]
```diff
@@ -14,8 +23,18 @@ import { MatCardModule } from '@angular/material/card';
...
+  declarations: [MovieViewComponent,
+    MovieDisplayAvailabilitiesComponent,
+    MovieDisplayProductionComponent,
+    MovieDisplayPrincipalInfoComponent,
+    MovieDisplaySynopsisComponent,
+    MovieDisplayFilmDetailsComponent,
+    MovieDisplayPrizesComponent,
+    MovieDisplayAssetsComponent,
+    MovieDisplayKeywordsComponent,
+    MovieDisplayVersionInfoComponent],
```
Comment (Maxstgt) [inline apps/catalog-marketplace/catalog-marketplace/src/app/movie/view/view.module.ts]: Import the modules from the components into the imports array. I think, if you now import the MovieDisplayPrizesComponent somewhere else, you will get an error that says: This component is declared in to modules... maybe try it out.

File: libs/movie/src/lib/movie/components/display-film-details/display-film-details.component.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Component, ChangeDetectionStrategy, Input } from "@angular/core";
+
+@Component({
+  selector: '[movie] movie-display-film-details',
+  templateUrl: './display-film-details.component.html',
+  styleUrls: ['./display-film-details.component.scss'],
+  changeDetection: ChangeDetectionStrategy.Default
```
Comment (Maxstgt) [inline libs/movie/src/lib/movie/components/display-film-details/display-film-details.component.ts]: why not OnPush?

File: libs/movie/src/lib/movie/components/display-film-details/display-film-details.module.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { NgModule } from "@angular/core";
+import { MovieDisplayFilmDetailsComponent } from "./header.component";
+import { CommonModule } from "@angular/common";
+
+@NgModule({
+  imports: [CommonModule]
+  declarations: [MovieDisplayFilmDetailsComponent],
```
Comment (Maxstgt) [inline libs/movie/src/lib/movie/components/display-film-details/display-film-details.module.ts]: @pldespaigne should the compoent take care of the imports of the Material modules or should do this the page/parent component?

File: libs/movie/src/lib/movie/components/display-version-info/display-version-info.component.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Component, ChangeDetectionStrategy, Input } from "@angular/core";
+
+@Component({
+  selector: '[movie] movie-display-version-info',
+  templateUrl: './display-version-info.component.html',
+  styleUrls: ['./display-version-info.component.scss'],
+  changeDetection: ChangeDetectionStrategy.Default
```
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-version-info/display-version-info.component.ts]: use `OnPush`

File: libs/movie/src/lib/movie/components/display-synopsis/display-synopsis.component.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { Component, ChangeDetectionStrategy, Input } from "@angular/core";
+
+@Component({
+  selector: '[movie] movie-display-synopsis',
+  templateUrl: './display-synopsis.component.html',
+  styleUrls: ['./display-synopsis.component.scss'],
+  changeDetection: ChangeDetectionStrategy.Default
```
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-synopsis/display-synopsis.component.ts]: use `OnPush`

File: libs/movie/src/lib/movie/components/display-production/display-production.component.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { Component, ChangeDetectionStrategy, Input } from "@angular/core";
+
+@Component({
+  selector: '[movie] movie-display-production',
+  templateUrl: './display-production.component.html',
+  styleUrls: ['./display-production.component.scss'],
+  changeDetection: ChangeDetectionStrategy.Default
```
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-production/display-production.component.ts]: use `OnPush`

File: libs/movie/src/lib/movie/components/display-prizes/display-prizes.component.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import { Component, ChangeDetectionStrategy, Input } from "@angular/core";
+
+@Component({
+  selector: '[movie] movie-display-prizes',
+  templateUrl: './display-prizes.component.html',
+  styleUrls: ['./display-prizes.component.scss'],
+  changeDetection: ChangeDetectionStrategy.Default
```
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-prizes/display-prizes.component.ts]: use `OnPush`

File: libs/movie/src/lib/movie/components/display-principal-info/display-principal-info.component.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import { Component, ChangeDetectionStrategy, Input } from "@angular/core";
+
+@Component({
+  selector: '[movie] movie-display-principal-info',
+  templateUrl: './display-principal-info.component.html',
+  styleUrls: ['./display-principal-info.component.scss'],
+  changeDetection: ChangeDetectionStrategy.Default
```
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-principal-info/display-principal-info.component.ts]: use `OnPush`

File: libs/movie/src/lib/movie/components/display-keywords/display-keywords.component.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import { Component, ChangeDetectionStrategy, Input } from "@angular/core";
+
+@Component({
+  selector: '[movie] movie-display-keywords',
+  templateUrl: './display-keywords.component.html',
+  styleUrls: ['./display-keywords.component.scss'],
+  changeDetection: ChangeDetectionStrategy.Default
```
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-keywords/display-keywords.component.ts]: use `OnPush`

File: libs/movie/src/lib/movie/components/display-film-details/display-film-details.component.ts (hunk 3 above)
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-film-details/display-film-details.component.ts]: use `OnPush`

File: libs/movie/src/lib/movie/components/display-availabilities/display-availabilities.component.ts [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import { Component, ChangeDetectionStrategy, Input } from "@angular/core";
+
+@Component({
+  selector: '[movie] movie-display-availabilities',
+  templateUrl: './display-availabilities.component.html',
+  styleUrls: ['./display-availabilities.component.scss'],
+  changeDetection: ChangeDetectionStrategy.Default
```
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-availabilities/display-availabilities.component.ts]: use `OnPush`

File: libs/movie/src/lib/movie/components/display-assets/display-assets.component.ts [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import { Component, ChangeDetectionStrategy, Input } from "@angular/core";
+
+@Component({
+  selector: '[movie] movie-display-assets',
+  templateUrl: './display-assets.component.html',
+  styleUrls: ['./display-assets.component.scss'],
+  changeDetection: ChangeDetectionStrategy.Default
```
Comment (pldespaigne) [inline libs/movie/src/lib/movie/components/display-assets/display-assets.component.ts]: use `OnPush`

**Comment by jenna1k** [comment]
> Well done, great job !
> There is just the module problem to correct otherwise it is perfect. Did you use the cli to generate your components ?
> 
> The other thing is the `ChangeDetectionStrategy`, we prefer to use `OnPush` instead of `Default`, [here](https://alligator.io/angular/change-detection-strategy/) you can read a good explanation of why.

File: apps/catalog-marketplace/catalog-marketplace/src/app/movie/view/view.module.ts [hunk 13]
```diff
@@ -14,8 +23,18 @@ import { MatCardModule } from '@angular/material/card';
 import { MatDividerModule } from '@angular/material';
 import { MatButtonModule } from '@angular/material/button';
 
+
 @NgModule({
-  declarations: [MovieViewComponent],
+  declarations: [MovieViewComponent,
```
[the remark this answers is not in the document]
Comment (jenna1k) [reply apps/catalog-marketplace/catalog-marketplace/src/app/movie/view/view.module.ts]: how should I break lines here?

File: apps/catalog-marketplace/catalog-marketplace/src/app/movie/view/view.module.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jenna1k) [reply apps/catalog-marketplace/catalog-marketplace/src/app/movie/view/view.module.ts]: Max recommended me to create module for each component and I checked `MovieDisplayCardItemModule` as a reference

**Comment by jenna1k** [comment]
@pldespaigne thanks for the link about `OnPush`!
I didn't use cli for creating components and modules

File: apps/catalog-marketplace/catalog-marketplace/src/app/movie/view/view.module.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (pldespaigne) [reply apps/catalog-marketplace/catalog-marketplace/src/app/movie/view/view.module.ts]:
```
declarations: [
  blabla,
  blabla,
],
```

**Pull request merged**